service/builder/pipelines_builder.go

```diff
@@ -45,33 +46,33 @@ type builtPipeline struct {
 // BuiltPipelines is a map of build pipelines created from pipeline configs.
 type BuiltPipelines map[*configmodels.Pipeline]*builtPipeline
 
-func (bps BuiltPipelines) StartProcessors(logger *zap.Logger, host component.Host) error {
-	for cfg, bp := range bps {
-		logger.Info("Pipeline is starting...", zap.String("pipeline", cfg.Name))
+func (bps BuiltPipelines) StartProcessors(ctx context.Context, host component.Host) error {
+	for _, bp := range bps {
+		bp.logger.Info("Pipeline is starting...")
```

Comment:
And this lost the pipeline name?

Reply:
The logger in the builtPipeline has attached the pipeline name. See https://github.com/open-telemetry/opentelemetry-collector/pull/829/files/1bedf14b1dfb74f4176d1f2991042944e5925597#diff-5e0eb252114499cab694f83fef1492f9R186